citrination_client/client.py

```diff
@@ -55,6 +65,116 @@ def search(self, pif_query):
         return PifSearchResult(**keys_to_snake_case(response.json()['results']))
 
+    def simple_chemical_search(self, name=None, chemical_formula=None, property_name=None, property_value=None,
+                               property_min=None, property_max=None, property_units=None, reference_doi=None,
+                               include_datasets=[], exclude_datasets=[], from_index=None, size=None):
+        """
+        Run a query against Citrination. This method generates a :class:`.PifQuery` object using the supplied arguments.
+        All arguments that accept lists have logical OR's on the queries that they generate. This means that, for
+        example, simple_chemical_search(name=['A', 'B']) will match records that have name equal to 'A' or 'B'.
+
+        Results will be pulled into the extracted field of the :class:`.PifSearchHit` objects that are returned. The
+        name will appear under the key "name", chemical formula under "chemical_formula", property name under
+        "property_name", value of the property under "property_value", units of the property under "property_units",
+        and reference DOI under "reference_doi".
+
+        This method is only meant for execution of very simple queries. More complex queries must use the search method
+        that accepts a :class:`.PifQuery` object.
+
+        :param name: One or more strings with the names of the chemical system to match.
+        :param chemical_formula:  One or more strings with the chemical formulas to match.
+        :param property_name: One or more strings with the names of the property to match.
+        :param property_value: One or more strings or numbers with the exact values to match.
+        :param property_min: A single string or number with the minimum value to match.
+        :param property_max: A single string or number with the maximum value to match.
+        :param property_units: One or more strings with the property units to match.
+        :param reference_doi: One or more strings with the DOI to match.
+        :param include_datasets: One or more integers with dataset IDs to match.
+        :param exclude_datasets: One or more integers with dataset IDs that must not match.
+        :param from_index: Index of the first record to match.
+        :param size: Total number of records to return.
+        :return: :class:`.PifSearchResult` object with the results of the query.
+        """
+        system_query = SystemQuery()
+
+        # Add the name operation
+        if isinstance(name, list):
+            system_query.names = [FieldOperation(extract_as='name', filter=Filter(equal=i)) for i in name]
+        elif isinstance(name, string_types):
+            system_query.names = [FieldOperation(extract_as='name', filter=Filter(equal=name))]
+        else:
+            system_query.names = [FieldOperation(extract_as='name')]
```

Comment:
This pattern of 
```
if list, then then return list of reshaped items
else if string then return list of reshaped strings
else return reshaped thing
```
seems to repeat through the rest of the this method.  Extracting it as a function would reduce both the duplication and the obfuscation of the high-level "recipe" that results from the `if/elif/else` blocks.